packages/@aws-cdk/custom-resource-handlers/lib/aws-s3/notifications-resource-handler/index.py

```diff
@@ -48,6 +48,10 @@ def with_id(n):
         old_incoming_ids = [n['Id'] for n in ids]
         # if the notification was created by us, we know what id to expect so we can filter by it.
         external_notifications[t] = [n for n in existing_notifications.get(t, []) if not n['Id'] in old_incoming_ids]
+    elif request_type == 'Delete':
+        # For 'Delete' request, old parameter is an empty dict so we cannot use this to determine which are external
+        # notifications. Fall back to rely on the stack naming logic.
+        external_notifications[t] = [n for n in existing_notifications.get(t, []) if not n['Id'].startswith(f"{stack_id}-")]
```

Comment:
Falling back to the previous correct solution https://github.com/aws/aws-cdk/commit/37be7b9b5d2abe04aa5f632a7b78aed3e9c233d8#diff-00d1f29e73955f487795d4a9873bcff824e70d6268e7926e44b7defb31b4ecb1L74